core/controllers/async_download.py

```diff
@@ -0,0 +1,112 @@
+import io
+import uuid
+from typing import IO
+
+from flask import current_app
+from notifications_python_client.notifications import NotificationsAPIClient
+from werkzeug.datastructures import FileStorage
+
+from config import Config
+from core.aws import _S3_CLIENT
+from core.controllers.download import download
+
+
+def async_download(
+    email_address: str,
+    file_format: str,
+    funds: list[str] | None = None,
+    organisations: list[str] | None = None,
+    regions: list[str] | None = None,
+    rp_start: str | None = None,
+    rp_end: str | None = None,
+    outcome_categories: list[str] | None = None,
+):
+    """Download data, store file in S3 and send an email to the user with the download link."""
+
+    response = download(
+        file_format=file_format,
+        funds=funds,
+        organisations=organisations,
+        regions=regions,
+        rp_start=rp_start,
+        rp_end=rp_end,
+        outcome_categories=outcome_categories,
+    )
+
+    file_format_str: str = ""
+    if file_format == "xlsx":
+        file_format_str = "Microsoft Excel spreadsheet"
+    elif file_format == "json":
+        file_format_str = "JSON file"
+
+    file_size_kb = round(len(response.data) / 1024, 1)
+    if file_size_kb < 1024:
+        file_size_str: str = f"{round(file_size_kb, 1)} KB"
+    else:
+        file_size_str = f"{round(file_size_kb / 1024, 1)} MB"
```

Comment:
These could probably be extracted to small helper functions: `get_file_format_from_extension` and `get_human_readable_file_size` or something similar?

Reply:
Separated to helper functions.